metrics/prometheus.go

```diff
@@ -1079,6 +1079,23 @@ func NewPrometheusCollector(i infoProvider, f ContainerLabelsFunc, includedMetri
 					}
 				},
 			},
+			{
+				name:        "container_root_process_ulimits_soft",
+				help:        "Soft ulimit values for the container root process.",
+				valueType:   prometheus.GaugeValue,
+				extraLabels: []string{"ulimit_name"},
+				getValues: func(s *info.ContainerStats) metricValues {
+					values := make(metricValues, 0, len(s.Processes.Ulimits))
+					for _, ulimit := range s.Processes.Ulimits {
```

Comment:
@dashpole : this will generate 16 metrics for each container. if that's too much I can scope it down to just file descriptors for now, and people can emit other ulimits as and when they're needed.

Reply:
Yeah, that is a lot.  I would prefer keeping the prometheus metrics the same as the json metrics, though.  If we do filtering, lets do it at collection time.

Usually, it starts out as a constant whitelist defined in the file.  We may need to add it as a flag eventually, but we can start with it hard-coded.

Reply:
Can we add a whitelist to container/libcontainer/handler.go to only store ulimit for FD limits?
Or it can include others if you think more are commonly used.

Reply:
Sorry about that, I didn't fully follow what you meant the first time around. Added a whitelist array to handler.go